### PR TITLE
TM-44 New detekt rules based on feedback

### PR DESCRIPTION
**Pull request by Schife**
Trying to remove some of the noise caused by a few rules in the test files.
Large class default has been increased to 600 in Detekt. Adjusted to new default.
Raised TooManyFunctions threshold so that builder classes are not affected anymore

### CODE REVIEW

File: detekt-config.yml [hunk 1]
```diff
@@ -32,6 +9,7 @@ build:
 processors:
   active: true
   exclude:
+  # - 'DetektProgressListener'
```
Comment (zkiss) [inline detekt-config.yml]: Any reason we have these comments here? I can't see the point.

**Review by zkiss — DISMISSED** [review]

File: detekt-config.yml (hunk 1 above)
Comment (Schife) [reply detekt-config.yml]: Honestly me neither. They look like extra things that we can enable during a run to display additional information. And are just listed there so people are aware of them. If people agree we can just strip the processors and console-reports away from the config. And I would go as far as stripping inactive rules as well

Comment (zkiss) [reply detekt-config.yml]: We can read the docs if we need to enable functionality, I vote to remove things that are not in play

**Review by r3domfox — APPROVED** [review]
LGTM

**Pull request merged**